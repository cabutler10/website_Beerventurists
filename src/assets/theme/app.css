html {
  height: 100%;
}

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  background-color: #ffffff;
  overflow-x: hidden;
  font-size: 14px;
  width: '100%';
}

a {
  text-decoration: none;
}
