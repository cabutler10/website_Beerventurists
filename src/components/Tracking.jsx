import React, { Fragment } from 'react'
import { connect } from 'react-redux'
import { Link } from 'gatsby'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/styles'
import Button from '@material-ui/core/Button'
import Snackbar from '@material-ui/core/Snackbar'
import SnackbarContent from '@material-ui/core/SnackbarContent'

const mapStateToProps = ({ isGAopen }) => {
  return { isGAopen }
}

const mapDispatchToProps = dispatch => {
  return { changeGA: () => dispatch({ type: `GOOGLEANALYTICS` }) }
}

const useStyles = makeStyles(theme => ({
  snackbar: {
    left: 0,
    right: 0,
    transform: 'none',
  },
  snackbarContent: {
    borderRadius: 0,
    minWidth: '0',
    maxWidth: 'none',
    color: theme.palette.common.black,
    backgroundColor: theme.status.greyBlue,
    [theme.breakpoints.up('sm')]: {
      width: '100%',
      paddingLeft: 30,
      paddingRight: 30,
    },
    [theme.breakpoints.up('md')]: {
      paddingLeft: 60,
      paddingRight: 60,
    },
  },
  message: {
    flexBasis: '100%',
    color: theme.palette.common.black,
    [theme.breakpoints.up('sm')]: {
      flexBasis: '60%',
    },
  },
  action: {
    marginLeft: 'none',
    marginRight: 'none',
    paddingLeft: 0,
  },
  button: {
    marginLeft: 30,
    paddingLeft: 30,
    paddingRight: 30,
    [theme.breakpoints.up('sm')]: {
      marginLeft: 35,
      paddingLeft: 50,
      paddingRight: 50,
    },
  },
  buttonDecline: {
    color: theme.palette.common.black,
  },
  link: {
    color: theme.palette.common.black,
    paddingLeft: 8,
    textTransform: 'capitalize',
    textDecoration: 'underline',
    [theme.breakpoints.up('sm')]: {
      paddingLeft: 0,
    },
    [theme.breakpoints.up('md')]: {
      paddingLeft: 8,
    },
  },
}))

function Tracking({ isGAopen, changeGA }) {
  const classes = useStyles()
  const { t } = useTranslation()

  const handleSnackbarDecline = () => {
    window['ga-disable-UA-130658859-1'] = true
    changeGA()
  }

  return (
    <Snackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      open={isGAopen}
      className={classes.snackbar}
    >
      <SnackbarContent
        className={classes.snackbarContent}
        classes={{ message: classes.message, action: classes.action }}
        aria-describedby="client-snackbar"
        message={
          <span id="client-snackbar">
            {t('tracking.message')}
            <Link to="/privacy/" className={classes.link}>
              {t('tracking.privacyLink')}
            </Link>
          </span>
        }
        action={[
          <Fragment key={`snackbar`}>
            <Button
              className={classes.buttonDecline} //eslint-disable-next-line
              href="javascript:gaOptout();"
              onClick={handleSnackbarDecline}
            >
              {t('tracking.decline')}
            </Button>
            <Button
              color="primary"
              variant="contained"
              className={classes.button}
              onClick={changeGA}
            >
              {t('tracking.accept')}
            </Button>
          </Fragment>,
        ]}
      />
    </Snackbar>
  )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Tracking)
