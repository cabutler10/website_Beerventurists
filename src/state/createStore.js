import { createStore as reduxCreateStore } from 'redux'
import i18n from 'i18next'

export const reducer = (state, action) => {
  switch (action.type) {
    case 'LANGUAGE':
      i18n.changeLanguage(state.lng === 'en' ? 'de' : 'en')
      return {
        lng: state.lng === 'en' ? 'de' : 'en',
        isGAopen: state.isGAopen,
      }
    case 'GOOGLEANALYTICS':
      return {
        isGAopen: false,
        lng: state.lng,
      }
    case 'PRODUCT':
      return {
        product: {},
        isGAopen: state.isGAopen,
        lng: state.lng,
      }
    default:
      return state
  }
}

const initialState = { lng: 'en', isGAopen: true, product: null }

const createStore = () => reduxCreateStore(reducer, initialState)

export default createStore
