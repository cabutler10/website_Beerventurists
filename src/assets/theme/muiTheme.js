import { createMuiTheme } from '@material-ui/core/styles'
import 'typeface-open-sans'
import 'typeface-permanent-marker'
import './app.css'

const status = {
  white: '#ffffff',
  black: '#040001',
  greyLt: '#F7F5F5',
  greyMed: '#656565',
  grey: '#c5c6c7',
  greyDk: '#404040',
  green: '#00942b',
  yellow: '#fce566',
  red: '#FF0012',
  blue: '#015b99',
  blueLt: '#0172c0',
  blueDk: '#014472',
  orange: '#c56113',
  orangeDk: '#c56113',
  orangeLt: '#ebab2f',
  tan: '#FFFBEF',
}

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: status.blueLt,
      main: status.blue,
      dark: status.blueDk,
    },
    secondary: {
      light: status.orangeLt,
      main: status.orange,
      dark: status.orangeDk,
    },
    error: { main: status.red },
  },
  status,
  typography: {
    useNextVariants: true,
    fontFamily: "'Open Sans', sans-serif",
    h5: {
      letterSpacing: 3,
    },
    h2: {
      fontFamily: "'Permanent Marker', cursive",
    },
    body1: {
      letterSpacing: 1,
      lineHeight: 1.8,
      //TODO is this an error in MUI? not sure why its default is 16 when MUI default is 14
      fontSize: 14,
    },
    body2: {
      letterSpacing: 1,
      fontSize: 14,
    },
    button: {
      letterSpacing: 3,
    },
  },
  overrides: {
    MuiTypography: {
      colorPrimary: {
        color: status.black,
      },
      colorSecondary: {
        color: status.orangeLt,
      },
      colorTextSecondary: {
        color: status.white,
      },
    },
    MuiButton: {
      root: {
        padding: '9px 27px',
        borderRadius: 0,
      },
      contained: {
        boxShadow: 'none',
      },
    },
  },
})

export default theme
