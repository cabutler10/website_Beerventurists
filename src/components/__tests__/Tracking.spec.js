import React from 'react'
import { render } from 'react-testing-library'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { reducer } from '../../state/createStore'
import Tracking from '../Tracking'
import { ThemeProvider } from '@material-ui/styles'
import theme from '../../assets/theme/muiTheme'

describe('Tracking', () => {
  it('renders correctly', () => {
    const mockProps = {
      t: () => '',
    }
    const store = createStore(reducer, { isGAopen: true })
    const component = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Tracking {...mockProps} />
        </ThemeProvider>
      </Provider>
    )
    expect(component).toMatchSnapshot()
  })
})
