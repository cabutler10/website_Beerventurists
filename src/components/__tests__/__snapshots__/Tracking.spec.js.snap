// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracking renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiSnackbar-root-1 MuiSnackbar-anchorOriginBottomCenter-3 Hook-snackbar-1x73or3"
      >
        <div
          aria-describedby="client-snackbar"
          class="MuiTypography-root-38 MuiTypography-body1-47 MuiPaper-root-11 MuiPaper-elevation6-19 MuiSnackbarContent-root-8 Hook-snackbarContent-vntwp2"
          role="alertdialog"
          style="-webkit-transform: translate(0, 0); transform: translate(0, 0); -webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiSnackbarContent-message-9 Hook-message-1f4qzjh"
          >
            <span
              id="client-snackbar"
            >
              tracking.message
              <a
                class="Hook-link-1ajxyk9"
                href="/privacy/"
              >
                tracking.privacyLink
              </a>
            </span>
          </div>
          <div
            class="MuiSnackbarContent-action-10 Hook-action-1w0eo5p"
          >
            <a
              class="MuiButtonBase-root-100 MuiButton-root-74 MuiButton-text-76 MuiButton-flat-79 Hook-buttonDecline-13ddspg"
              href="javascript:gaOptout();"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label-75"
              >
                tracking.decline
              </span>
              <span
                class="MuiTouchRipple-root-103"
              />
            </a>
            <button
              class="MuiButtonBase-root-100 MuiButton-root-74 MuiButton-contained-85 MuiButton-containedPrimary-86 MuiButton-raised-88 MuiButton-raisedPrimary-89 Hook-button-qulza8"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label-75"
              >
                tracking.accept
              </span>
              <span
                class="MuiTouchRipple-root-103"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiSnackbar-root-1 MuiSnackbar-anchorOriginBottomCenter-3 Hook-snackbar-1x73or3"
    >
      <div
        aria-describedby="client-snackbar"
        class="MuiTypography-root-38 MuiTypography-body1-47 MuiPaper-root-11 MuiPaper-elevation6-19 MuiSnackbarContent-root-8 Hook-snackbarContent-vntwp2"
        role="alertdialog"
        style="-webkit-transform: translate(0, 0); transform: translate(0, 0); -webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiSnackbarContent-message-9 Hook-message-1f4qzjh"
        >
          <span
            id="client-snackbar"
          >
            tracking.message
            <a
              class="Hook-link-1ajxyk9"
              href="/privacy/"
            >
              tracking.privacyLink
            </a>
          </span>
        </div>
        <div
          class="MuiSnackbarContent-action-10 Hook-action-1w0eo5p"
        >
          <a
            class="MuiButtonBase-root-100 MuiButton-root-74 MuiButton-text-76 MuiButton-flat-79 Hook-buttonDecline-13ddspg"
            href="javascript:gaOptout();"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label-75"
            >
              tracking.decline
            </span>
            <span
              class="MuiTouchRipple-root-103"
            />
          </a>
          <button
            class="MuiButtonBase-root-100 MuiButton-root-74 MuiButton-contained-85 MuiButton-containedPrimary-86 MuiButton-raised-88 MuiButton-raisedPrimary-89 Hook-button-qulza8"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-75"
            >
              tracking.accept
            </span>
            <span
              class="MuiTouchRipple-root-103"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
